### ISSUE TITLE
darktable crashes on loading a corrupt gpx file in the geotagging module of the map view

### ISSUE THREAD

**Issue by DanielFeld2020** [darktable-org/darktable#17426]
### Describe the bug

When loading a corrupt gpx file in the geotagging module of the map view darktable crashes. The corrupt gpx file was missing a `<trkseg>` element. Running darktable with -d all shows only  "Magick: caught exception 0xC0000005 "Access violation"..." in the created log file.

### Steps to reproduce

1. start darktable
2. go to map view
3. load corrupt gpx file, missing a `<trkseg>` element, in the geotagging module
4. darktable crashes

### Expected behavior

darktable should show an appropriate error message, if possible with the bonus of giving a hint WHAT is corrupt in the loaded gpx file

### Logfile | Screenshot | Screencast

version: darktable 4.8.1
start: 2024:09:04 10:44:41

Magick: caught exception 0xC0000005 "Access violation"...

### Where did you obtain darktable from?

downloaded from www.darktable.org

### darktable version

4.8.1

### What OS are you using?

Windows

### What is the version of your OS?

Windows 11 Home 23H2

### Are you using OpenCL GPU in darktable?

No

### Please provide additional context if applicable. You can attach files too, but might need to rename to .txt or .zip

Here is a stripped down corrupt gpx file, which is missing a `<trkseg> `element.

```
<?xml version="1.0" encoding="UTF-8"?>
<gpx version="1.0" creator="GPSBabel - https://www.gpsbabel.org" xmlns="http://www.topografix.com/GPX/1/0">
  <time>2024-09-01T12:56:22.042Z</time>
  <bounds minlat="28.639701337" minlon="-17.771033507" maxlat="28.688961918" maxlon="-17.758423576"/>
  <trk>
    <name>track-1</name>
    <desc>Log every 5 sec, 0 m</desc>
      <trkpt lat="28.651661061" lon="-17.760958358">
        <ele>77.607</ele>
        <time>2017-05-28T15:18:53Z</time>
        <speed>0.758234</speed>
        <name>TP000001</name>
        <fix>3d</fix>
      </trkpt>
    </trkseg>
  </trk>
</gpx>
```

**Comment by victoryforce** [comment]
Confirmed. How did you get the corrupted GPX file? It does not contain a `<trkseg>` element, but it does contain a matching closing element, `</trkseg>`. It seems that this is the result of manual editing, or the processing of an initially correct file by some buggy script.

But in any case, surely darktable should not crash even on corrupted files. A fix for this is in the works.

**Comment by zisoft** [comment]
@victoryforce: How far is your work on this? I am currently implementing the full error handling on the gpx parser.

**Comment by DanielFeld2020** [comment]
@victoryforce 
> Confirmed. How did you get the corrupted GPX file? [...]. It seems that this is the result of manual editing, or [...}

Yes, this is the result of manual editing! I'm currently loading old images from 2017 into darktable. Some of the images have gpx files that I already used for Lightroom back then. However, one of those gpx files I couldn't load into Lightroom and for whatever reason I was able to ‘fix’ the problem back then by removing some of the `<trkseg>` elements. Now, 7 years later, this ‘silly fix’ was my undoing . At least I had made a note of the fix and thus the problem in a text file...

> But in any case, surely darktable should not crash even on corrupted files.

That is the motivation for my bug report. Nothing urgent and nothing important, but every bug fixed helps darktable move forward.

> A fix for this is in the works.

Great! Thanks!

**Comment by victoryforce** [comment]
> @victoryforce: How far is your work on this? I am currently implementing the full error handling on the gpx parser.

@zisoft I just removed the possibility of NULL dereferencing and added handling of possible GPX parsing failure.

If you think that more should be added here (such as displaying an error message from the parser), then go ahead.

Although, to be honest, I believe that invalid files are an extremely rare situation and can only occur due to some manual intervention. So is it worth adding full error handling here?

**Comment by zisoft** [comment]
No problem, I just spent an hour to see what needs to be done, came back here and saw you are working on this.

Just implemented the GError handling to show a message and to prevent the crashes: [zisoft:darktable:gpx-error-handling](https://github.com/darktable-org/darktable/compare/master...zisoft:darktable:gpx-error-handling)

**Comment by victoryforce** [comment]
> I just spent an hour to see what needs to be done

I spent much less time, so better to make a PR from your fix.  Especially since my code is just a quick hack (or low-fidelity MVP, to put it a bit jokingly :), so I would have to spend some time before doing PR to recheck it.

> Just implemented the GError handling to show a message

I opened the error dialog, but now I think your approach with showing the notification pill is better.
